src/SM/StateMachine/StateMachine.php

```diff
@@ -212,20 +213,23 @@ protected function setState($state)
      * Builds and calls the defined callbacks
      *
      * @param TransitionEvent $event
-     * @param string          $position
+     * @param string $position
+     * @return bool
      */
     protected function callCallbacks(TransitionEvent $event, $position)
     {
+        $result = true;
         if (!isset($this->config['callbacks'][$position])) {
-            return;
+            return $result;
         }
 
         foreach ($this->config['callbacks'][$position] as &$callback) {
             if (!$callback instanceof CallbackInterface) {
                 $callback = $this->callbackFactory->get($callback);
             }
 
-            call_user_func($callback, $event);
+            $result = $result & call_user_func($callback, $event);
```

Comment:
Good point. My approach was to see guards as special callbacks, thus sharing the same execution logic. There are arguments for and against this approach.

In my opinion, guards are simple, lightweight condition checkers, which shouldn't have a performance impact or systemic side effects. Thus, I can tolerate that all of them are called, even if the first one already disabled the transition.

If you think heavier guards are more common, then we could easily break out of the `foreach` loop with something like
```
    if (!$result && $position == 'guards') {
        return false;
    }
```
What do you think is better?